python/example_code/lookoutvision/inference.py

```diff
@@ -28,47 +29,34 @@ class Inference:
     @staticmethod
     def detect_anomalies(lookoutvision_client, project_name, model_version, photo):
         """
-        Detects anomalies in an image (jpg/png) by using your Lookout for Vision model.
-
-        :param lookoutvision_client: A Boto3 Lookout for Vision client.
-        :param project_name: The name of the project that contains the model that
-                             you want to use.
+        Calls DetectAnomalies using the supplied project, model version, and image.
+        :param lookoutvision_client: A Lookout for Vision Boto3 client.
+        :param project: The project that contains the model that you want to use.
         :param model_version: The version of the model that you want to use.
-        :param photo: The path and name of the image in which you want to detect
-                      anomalies.
-        :return: Whether anomalies were detected and with what confidence.
+        :param photo: The photo that you want to analyze.
+        :return: The DetectAnomalyResult object that contains the analysis results.
         """
-        try:
-            image_type = imghdr.what(photo)
-            if image_type == "jpeg":
-                content_type = "image/jpeg"
-            elif image_type == "png":
-                content_type = "image/png"
-            else:
-                logger.info("Invalid image type for %s", photo)
-                raise ValueError(
-                    f"Invalid file format. Supply a jpeg or png format file: {photo}")
-
-            logger.info("Detecting anomalies in %s", photo)
-            with open(photo, "rb") as image:
-                response = lookoutvision_client.detect_anomalies(
-                    ProjectName=project_name,
-                    ContentType=content_type,
-                    Body=image.read(),
-                    ModelVersion=model_version)
-            anomalous = response["DetectAnomalyResult"]["IsAnomalous"]
-            confidence = response["DetectAnomalyResult"]["Confidence"]
-
-            logger.info("Anomalous?: %s", anomalous)
-            logger.info("Confidence: %s", confidence)
-        except FileNotFoundError:
-            logger.exception("Couldn't find file: %s", photo)
-            raise
-        except ClientError:
-            logger.exception("Couldn't detect anomalies.")
-            raise
+
+        image_type = imghdr.what(photo)
+        if image_type == "jpeg":
+            content_type = "image/jpeg"
+        elif image_type == "png":
+            content_type = "image/png"
         else:
-            return anomalous, confidence
+            logger.info("Invalid image type for %s", photo)
+            raise ValueError(
+                f"Invalid file format. Supply a jpeg or png format file: {photo}")
+
+
+        # Get images bytes for call to detect_anomalies
```

Comment:
# Get image bytes for call to detect_anomalies.

Reply:
Fixed.